falcon/util/misc.py

```diff
@@ -210,3 +212,25 @@ def get_bound_method(obj, method_name):
             raise AttributeError(msg)
 
     return method
+
+
+def get_http_status(status_code):
+    """Gets both the http status code and description from just a code
+
+    Args:
+        status_code: integer or string that can be converted to an integer
+
+    Returns:
+        str: status code e.g. "404 Not Found"
+
+    Raises:
+        ValueError: the value entered could not be converted to an integer
+
+    """
+    code = int(status_code)  # converts to integer or raises ValueError
+
+    try:
+        return getattr(falcon.status_codes, 'HTTP_' + str(code))
+    except AttributeError:
+        # not found
+        return str(code) + ' Unknown Status Code'
```

Comment:
w.r.t. to your 1), i meant to say that after doing an int, you do not need to convert it back to string, you can just re-use 'status_code'
And for '401.2', if you do int('401.2'), you would get:
>>> int('401.2')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: invalid literal for int() with base 10: '401.2'